account_bank_statement_import_online/models/online_bank_statement_provider.py

```diff
@@ -344,6 +368,10 @@ def _get_statement_date(self, date_since, date_until):
         # NOTE: Statement date is treated by Odoo as start of period. Details
         #  - addons/account/models/account_journal_dashboard.py
         #  - def get_line_graph_datas()
+        tz = timezone(self.tz) if self.tz else utc
```

Comment:
don't need this code, because `statement_date_since` is an datetime without tz (withou tz == utc).
https://github.com/OCA/bank-statement-import/blob/12.0/account_bank_statement_import_online/models/online_bank_statement_provider.py#L199

```python
def _get_statement_date(self, date_since, date_until):
     date_since = date_since.astimezone(provider_tz)
     return date_since.date()
```

Reply:
Valid point, fixed!